Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -379,38 +387,42 @@ void CGPathAddArc(CGMutablePathRef path,
                   bool clockwise) {
     RETURN_IF(!path);
 
-    // Normalize the start angle so it's between 0 and 2*pi
-    startAngle = fmod(startAngle, 2.0f * M_PI);
-    if (startAngle < 0.0f) {
-        startAngle += 2.0f * M_PI;
+    CGPoint startPoint = CGPointMake(x + radius * cos(startAngle), y + radius * sin(startAngle));
+    CGPoint endPoint = CGPointMake(x + radius * cos(endAngle), y + radius * sin(endAngle));
+
+    // Create the parameters for the AddArc method.
+    const D2D1_POINT_2F endPointD2D = _CGPointToD2D_F(endPoint);
+    const D2D1_SIZE_F radiusD2D = { radius, radius };
+    CGFloat rotationAngle = abs(startAngle - endAngle);
+    D2D1_ARC_SIZE arcSize = D2D1_ARC_SIZE_SMALL;
+    CGFloat expectedAngle = (clockwise ? startAngle + rotationAngle : startAngle - rotationAngle);
+
```

Comment:
The entire D2D section is very similar the one used in the function above.  it would be good to have this in a utility function and call it in both places.